### PR TITLE
Add "site flags" mechanism and convert 'schedule_published' to it

### PR DESCRIPTION
**Pull request by encukou**
For: https://github.com/pyvec/cz.pycon.org-2019/issues/5

- Add pyconcz.common.models.SiteFlag, a boolean that can be turned
  on/off at a specific time.
- Add SiteFlag to the admin
- Add a context processor to get all active site flags, and
  register it
- Add migration that creates a 'schedule_published' flag (off by
  default)
- Modify templates to use this flag, rather than overriding based
  on CURRENT_PHASE

### CODE REVIEW

**Comment by encukou** [comment]
Jako neDjangista prosím Djangistu o review – bourám best practices? @starenka?
Funkčnost dořešíme v https://github.com/pyvec/cz.pycon.org-2019/issues/5

**Comment by frenzymadness** [comment]
At first look, it looks good to me. But I am not the right person to do a review. We can also deploy it and test it.

File: pyconcz/common/models.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+from django.utils import timezone
+
+
+class SiteFlag(models.Model):
+    name = models.CharField(max_length=200, unique=True)
+    value_type = models.CharField(
+        max_length=200,
+        choices=[
+            # Human-readable titles are the same as the value
+            (v, v) for v in ("on", "off", "on_at", "off_at")
```
Comment (starenka) [inline pyconcz/common/models.py]:
```suggestion
            (v, v) for v in ("on", "off", "on_at", "off_at")
```
ty choices bejva fajn mit jako "konstantu" na ty tride (kdyz se to meni, tak se to meni na jednom miste, pouziva se to pak z klasy, je to takovy vic jasny, co to dela), ale nevidim to jako velkej trabl....

neco ala:

VALUE_TYPE_ON = 'on',  nebo se na to pouzivaj libky ala `django-choices`

File: pyconcz/common/models.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    def __str__(self):
+        return self.name
+
+    class Meta:
+        ordering = ('name', )
+
+    @classmethod
+    def get_active_flags(cls):
+        now = timezone.now()
+        return [obj.name for obj in (
```
Comment (starenka) [inline pyconcz/common/models.py]: tady bejva fajn si poslat "jen to, co potrebujes", tedy `cls.objects.filter(value_type='on').values_list('name')` 
jinak by asi bylo pro "djangisty" citelnejsi pouzit Q https://docs.djangoproject.com/en/2.1/topics/db/queries/#complex-lookups-with-q-objects misto tohohole chainu....

File: pyconcz/common/admin.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+from django.contrib import admin
+
+from .models import SiteFlag
+
+
+class SiteFlagAdmin(admin.ModelAdmin):
+    fields = ['name', 'value_type', 'switch_time']
+    list_display = ['name', 'value_type', 'switch_time', 'current_value']
+
```
Comment (starenka) [inline pyconcz/common/admin.py]: pro pohodlnost mozna neco hodit do [list_editable](https://docs.djangoproject.com/en/2.1/ref/contrib/admin/#django.contrib.admin.ModelAdmin.list_editable)

**Comment by starenka** [comment]
ty komenty sou drobnosti, jestli to funguje, tak citelny je to v pohode, neresil bych to

**Comment by frenzymadness** [comment]
@benabraham anything against merging this? IMHO this is much better than the current state/workflow.

**Comment by benabraham** [comment]
We don’t need that **right** now (everything will be in a desired state for some time), so feel free to refactor as needed and merge after.

**Comment by frenzymadness** [comment]
Dan is right. @encukou if you want to address comments, feel free to do it. But if you don't have time to do so, I'll merge it as is next weekend when we'll have some time to play with it.

**Comment by frenzymadness** [comment]
Next changes on website will be:

* keynoters
* speakers
* programme

@benabraham could you please consider fixing the conflict here sooner than later so we can use this features?

**Comment by encukou** [comment]
I am planning to address the comments, but I'll only get to it on Friday.

**Comment by encukou** [comment]
Possibly superseded by #85.

**Comment by benabraham** [comment]
Definitely superseeded by #85, let's use that!

**Pull request closed without merging**